lms/djangoapps/course_structure_api/v0/tests.py

```diff
@@ -0,0 +1,334 @@
+
+        courses = response.data['results']
+        self.assertEqual(len(courses), 1)
+        self.assertValidResponseCourse(courses[0], self.empty_course)
+
+    def test_get_filtering(self):
+        """
+        The view should return a list of details for the specified courses.
+        """
+        url = "{}?course_id={}".format(reverse(self.view), self.course_id)
+        response = self.http_get(url)
+        self.assertEqual(response.status_code, 200)
+
+        courses = response.data['results']
+        self.assertEqual(len(courses), 1)
+        self.assertValidResponseCourse(courses[0], self.course)
+
+    def test_not_authenticated(self):
+        # If debug mode is enabled, the view should always return data.
+        with override_settings(DEBUG=True):
+            response = self.http_get(reverse(self.view), HTTP_AUTHORIZATION=None)
+            self.assertEqual(response.status_code, 200)
+
+        response = self.http_get(reverse(self.view), HTTP_AUTHORIZATION=None)
+        self.assertEqual(response.status_code, 401)
+
+    def test_not_authorized(self):
+        """
+        Unauthorized users should get an empty list.
+        """
+        user = StaffFactory(course_key=self.course.id)
+        access_token = AccessTokenFactory.create(user=user, client=self.oauth_client).token
+        auth_header = 'Bearer ' + access_token
+
+        # If debug mode is enabled, the view should always return data.
+        with override_settings(DEBUG=True):
+            response = self.http_get(reverse(self.view), HTTP_AUTHORIZATION=auth_header)
+            self.assertEqual(response.status_code, 200)
+
+        # Data should be returned if the user is authorized.
+        response = self.http_get(reverse(self.view), HTTP_AUTHORIZATION=auth_header)
+        self.assertEqual(response.status_code, 200)
+
+        url = "{}?course_id={}".format(reverse(self.view), self.course_id)
+        response = self.http_get(url, HTTP_AUTHORIZATION=auth_header)
+        self.assertEqual(response.status_code, 200)
+        data = response.data['results']
+        self.assertEqual(len(data), 1)
+        self.assertEqual(data[0]['name'], self.course.display_name)
+
+        # The view should return an empty list if the user cannot access any courses.
+        url = "{}?course_id={}".format(reverse(self.view), unicode(self.empty_course.id))
+        response = self.http_get(url, HTTP_AUTHORIZATION=auth_header)
+        self.assertEqual(response.status_code, 200)
+        self.assertDictContainsSubset({'count': 0, u'results': []}, response.data)
+
+
+@override_settings(MODULESTORE=TEST_DATA_MOCK_MODULESTORE)
+class CourseDetailTests(CourseDetailMixin, CourseViewTestsMixin, ModuleStoreTestCase):
+    view = 'course_structure_api:v0:detail'
+
+    def test_get(self):
+        response = super(CourseDetailTests, self).test_get()
+        self.assertValidResponseCourse(response.data, self.course)
+
+
+@override_settings(MODULESTORE=TEST_DATA_MOCK_MODULESTORE)
```

Comment:
This has changed since you set this PR up. For `TEST_DATA_MOCK_MODULESTORE`, you should just remove the `override_settings` entirely. If you need a specific `MODULESTORE`, you should set it as a class attribute.